src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -200,12 +230,17 @@ public Task Stop()
 
                     if (!terminated && !_requestProcessingStopping)
                     {
-                        MessageBody = MessageBody.For(HttpVersion, _requestHeaders, this);
-                        _keepAlive = MessageBody.RequestKeepAlive;
-                        RequestBody = new FrameRequestStream(MessageBody);
-                        ResponseBody = new FrameResponseStream(this);
+                        var messageBody = MessageBody.For(HttpVersion, _requestHeaders, this);
+                        _keepAlive = messageBody.RequestKeepAlive;
+                        _requestBody = new FrameRequestStream(messageBody);
+                        RequestBody = _requestBody;
+                        _responseBody = new FrameResponseStream(this);
+                        ResponseBody = _responseBody;
                         DuplexStream = new FrameDuplexStream(RequestBody, ResponseBody);
 
+                        _requestAbortCts = CancellationTokenSource.CreateLinkedTokenSource(_disconnectCts.Token);
```

Comment:
I've found that there isn't a measurable perf hit if no one registers to the RequestAborted token.

When running the wrk against https://github.com/aspnet/benchmarks' plaintext endpoint and registering with the RequestAborted token on each request there is about a 10% decrease in throughput.